### PR TITLE
Fix some broken code in master

### PR DESCRIPTION
**Pull request by mkeeler**
There were several PRs that while all passed CI independently, when they all got merged into the same branch caused compilation errors in test code.

The main changes that caused issues where changing agent/cache.Cache.New to require a concrete options struct instead of a pointer. This broke the cert monitor tests and the catalog_list_services_test.go. Another change was made to unembed the http.Server from the agent.HTTPServer struct. That coupled with another change to add a test to ensure cache rate limiting coming from HTTP requests was working as expected caused compilation failures.

### CODE REVIEW

File: agent/cache/cache.go [hunk 1]
```diff
@@ -35,6 +35,9 @@ import (
 const (
 	CacheRefreshBackoffMin = 3               // 3 attempts before backing off
 	CacheRefreshMaxWait    = 1 * time.Minute // maximum backoff wait time
+
+	defaultEntryFetchRate     = rate.Inf // don't rate limit
+	defaultEntryFetchMaxBurst = 1000     // this is well above anything that should be necessary
```
Comment (freddygv) [inline agent/cache/cache.go:40]: should this value diverge from the defaults in `agent/config/builder.go`?

Comment (dnephin) [reply agent/cache/cache.go:40]: I think if rate is `rate.Inf` the burst value is ignored, so we could use any value here.

Comment (mkeeler) [reply agent/cache/cache.go]: I went ahead and removed the constants from builder.go and just made builder.go reference the now exported consts from the cache package.

Comment (freddygv) [reply agent/cache/cache.go]: ah, yep: `As a special case, if r == Inf (the infinite rate), b is ignored.`

**Comment by i0rek** [comment]
👍

**Pull request merged**